package.json

```diff
@@ -64,6 +64,7 @@
     "grunt-jscs": "2.1.0",
     "grunt-mocha-cli": "1.14.0",
     "hawk": "3.1.0",
+    "halfpenny": "*",
```

Comment:
Firstly, thanks for having this long conversation with me. Sorry if I'm being a PITA. The conversation has helped me to think through all the ramifications of this change.

I think we're on the same page now. Here is the remaining crux of the issue:

We have three goals ATM: 

1. Allow the API Client (_halfpenny_) to be more easily included in other projects without including the entire API codebase.
1. The API's integration tests should still rely on _halfpenny_. This serves to test halfpenny and the API simultaneously.
1. All integration tests should run on a fresh clone of the **default branch** of the API after running standard post-clone commands (`npm i`, `grunt build`).

Currently, the default branch of the API is _develop_. This means that the version of _halfpenny_ that is listed in _package.json_ needs to be the version that works with the current state of the _develop_ branch of the API. 

I like the idea of publishing new versions of _halfpenny_ to NPM from the master branch of the _halfpenny_ repo during the build process. Unfortunately, this makes it difficult to maintain a pointer to the proper version of _halfpenny_ in the _develop_ branch of the API: There will be no release of the _halfpenny_ package until the API and _halfpenny_ are released to production. 

Two possible solutions come to mind:
1. Change the default branch of the API to _master_. This would mean that cloning the API would checkout the master branch, which should point to the latest version of _halfpenny_ that was published to NPM.
1. Use a github URL to reference the develop branch of _halfpenny_ in the API's _package.json_ file. This would cause NPM to install the latest version of the _halfpenny_ develop branch on NPM install.

What do you think of all that?

The core of the problem is that the nodeapi tests are completely dependent upon _halfpenny_, and _halfpenny_ has no tests of its own, so we can't modify it without testing it inside of the API... Essentially, it is a separation of concerns nightmare...




Our goal is to separate the API Client (halfpenny) from the API codebase so that it can be easily included in other repositories. In addition, the API tests will still rely on _halfpenny_, and will also function as tests of the API client itself. We want the API tests to run successfully when the default branch of the API is cloned and `npm install` is run.